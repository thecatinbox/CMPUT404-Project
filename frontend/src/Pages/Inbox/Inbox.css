.inbox {
    width: 60%;
    margin: auto;
}