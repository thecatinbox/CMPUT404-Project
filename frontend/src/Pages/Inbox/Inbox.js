import './Inbox.css';
import Message from '../../Components/Message/Message'; 
import TopBar from "../../Components/TopBar/TopBar";

const messageData = [
  {
    content: "Username1 liked your post"
  },
  {
    content: "Username2 wants to follow you"
  },
  {
    content: "Username3 commented on your post"
  },
];

function Inbox() {
  return (
    <>
      <TopBar id="inbox"/>
      <div className="inbox">
        {messageData.map(function(message){
            return <Message message={message}/>;
        })}
      </div>
    </>
  );
}

export default Inbox;