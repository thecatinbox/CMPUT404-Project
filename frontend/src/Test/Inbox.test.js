import Inbox from "../Pages/Inbox/Inbox";
import React from 'react';
import renderer from 'react-test-renderer';

it('inbox page renders correctly', () => {
    const component = renderer.create(
      <Inbox />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});