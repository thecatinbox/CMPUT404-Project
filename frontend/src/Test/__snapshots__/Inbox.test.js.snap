// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`inbox page renders correctly 1`] = `
Array [
  <div
    className="topnav"
  >
    <a
      href="/"
      id="home"
    >
      Home
    </a>
    <a
      href="/inbox"
      id="inbox"
    >
      Inbox
    </a>
    <a
      href="/friends"
      id="friends"
    >
      Friends
    </a>
    <a
      href="/profile"
      id="profile"
    >
      Profile
    </a>
    <div
      className="search-container"
    >
      <form
        action="/action_page.php"
      >
        <input
          name="search"
          placeholder="Search.."
          type="text"
        />
        <button
          type="submit"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-magnifying-glass "
            data-icon="magnifying-glass"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M416 208c0 45.9-14.9 88.3-40 122.7L502.6 457.4c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9 0 208S93.1 0 208 0S416 93.1 416 208zM208 352a144 144 0 1 0 0-288 144 144 0 1 0 0 288z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </button>
      </form>
    </div>
  </div>,
  <div
    className="inbox"
  >
    <div
      className="message"
    >
      <div
        className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-pwngrv-MuiPaper-root-MuiCard-root"
      >
        <div
          className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <p
            className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-envelope "
              data-icon="envelope"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M48 64C21.5 64 0 85.5 0 112c0 15.1 7.1 29.3 19.2 38.4L236.8 313.6c11.4 8.5 27 8.5 38.4 0L492.8 150.4c12.1-9.1 19.2-23.3 19.2-38.4c0-26.5-21.5-48-48-48H48zM0 176V384c0 35.3 28.7 64 64 64H448c35.3 0 64-28.7 64-64V176L294.4 339.2c-22.8 17.1-54 17.1-76.8 0L0 176z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
             
            Username1 liked your post
          </p>
        </div>
      </div>
    </div>
    <div
      className="message"
    >
      <div
        className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-pwngrv-MuiPaper-root-MuiCard-root"
      >
        <div
          className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <p
            className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-envelope "
              data-icon="envelope"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M48 64C21.5 64 0 85.5 0 112c0 15.1 7.1 29.3 19.2 38.4L236.8 313.6c11.4 8.5 27 8.5 38.4 0L492.8 150.4c12.1-9.1 19.2-23.3 19.2-38.4c0-26.5-21.5-48-48-48H48zM0 176V384c0 35.3 28.7 64 64 64H448c35.3 0 64-28.7 64-64V176L294.4 339.2c-22.8 17.1-54 17.1-76.8 0L0 176z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
             
            Username2 wants to follow you
          </p>
        </div>
      </div>
    </div>
    <div
      className="message"
    >
      <div
        className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-pwngrv-MuiPaper-root-MuiCard-root"
      >
        <div
          className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <p
            className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-envelope "
              data-icon="envelope"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M48 64C21.5 64 0 85.5 0 112c0 15.1 7.1 29.3 19.2 38.4L236.8 313.6c11.4 8.5 27 8.5 38.4 0L492.8 150.4c12.1-9.1 19.2-23.3 19.2-38.4c0-26.5-21.5-48-48-48H48zM0 176V384c0 35.3 28.7 64 64 64H448c35.3 0 64-28.7 64-64V176L294.4 339.2c-22.8 17.1-54 17.1-76.8 0L0 176z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
             
            Username3 commented on your post
          </p>
        </div>
      </div>
    </div>
  </div>,
]
`;
